src/plugins/procdump/procdump.cpp

```diff
@@ -0,0 +1,832 @@
+    auto ctx = new procdump_ctx;
+    ctx->pid = attached_proc_data.pid;
+    ctx->ppid = attached_proc_data.ppid;
+    ctx->tid = attached_proc_data.tid;
+    ctx->name = std::string(attached_proc_data.name);
+    ctx->plugin = plugin;
+    ctx->idx = plugin->procdumps_count++;
+    ctx->size = 0;
+    ctx->offset = 0;
+    // Get virtual address space map of the process
+    drakvuf_traverse_mmvad(drakvuf, attached_proc_data.base_addr, dump_mmvad,
+                           ctx);
+    if (ctx->vads.empty())
+    {
+        // nothing to do
+        delete ctx;
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+    {
+        PRINT_DEBUG("[PROCDUMP] [\"%s\":%4d] [PID:%4d] [TID:%4d] [\"%s\"] "
+                    "Dump 0x%lx (%ld MiB, %lu VADs) SN=%lu\n",
+                    __FUNCTION__, __LINE__, attached_proc_data.pid,
+                    attached_proc_data.tid, attached_proc_data.name, ctx->size,
+                    ctx->size / 1024 / 1024, ctx->vads.size(), ctx->idx);
+    }
+    ctx->file_path =
+        plugin->procdump_dir + "/procdump." + std::to_string(ctx->idx);
+    auto fd =
+        open((ctx->file_path + ".mm").data(), O_RDWR | O_CREAT | O_TRUNC, 0600);
+    if (-1 == fd)
+    {
+        PRINT_DEBUG("[PROCDUMP] Failed to create file\n");
+        delete ctx;
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+    else if (-1 == ftruncate(fd, ctx->size))
+    {
+        PRINT_DEBUG("[PROCDUMP] Failed to truncate file up to %zu\n",
+                    ctx->size);
+        delete ctx;
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+    else
+    {
+        ctx->file = static_cast<uint8_t*>(mmap(
+            nullptr, ctx->size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0));
+        if (MAP_FAILED == ctx->file)
+        {
+            PRINT_DEBUG("[PROCDUMP] Failed to mmap file\n");
+            delete ctx;
+            return VMI_EVENT_RESPONSE_NONE;
+        }
+    }
+    // Save registers to restore process/thread state
+    memcpy(&ctx->saved_regs, info->regs, sizeof(x86_registers_t));
+    ctx->bp = (drakvuf_trap_t*)g_try_malloc0(sizeof(drakvuf_trap_t));
+    ctx->pool = find_pool(plugin->pools);
+    ctx->bp->type = BREAKPOINT;
+    ctx->bp->cb = terminate_process_cb;
+    ctx->bp->data = ctx;
+    ctx->bp->breakpoint.lookup_type = LOOKUP_DTB;
+    ctx->bp->breakpoint.dtb = info->regs->cr3;
+    ctx->bp->breakpoint.addr_type = ADDR_VA;
+    ctx->bp->breakpoint.addr = info->regs->rip;
+    if (drakvuf_add_trap(drakvuf, ctx->bp))
+    {
+        bool is_continue = false;
+
+        if (ctx->pool)
+            is_continue = dump_next_vads(drakvuf, info, ctx);
+        else
+            is_continue = inject_allocate_pool(drakvuf, info, ctx);
+
+        if (is_continue)
+            return VMI_EVENT_RESPONSE_SET_REGISTERS;
+    }
+    else
+    {
+        PRINT_DEBUG("[PROCDUMP] Failed to trap return location of injected "
+                    "function call @ 0x%lx!\n",
+                    ctx->bp->breakpoint.addr);
```

Comment:
You should probably free `ctx->bp` here, the only place I spotted it getting freed is in its callback, which of course in this case would never get called.

Reply:
This is deleted in `detach` function.